### ISSUE TITLE
Force-closing Resonite when the cloud home is not fully loaded can 'scramble' textures in the cloud home (World Materials Seemingly Missing and Scrambled)

### ISSUE THREAD

**Issue by Nytra** [Yellow-Dog-Man/Resonite-Issues#280]
### Describe the bug?

When you close Resonite using the red X in the top right of the window in desktop mode, it will try to save homes before closing.

If you try to close Resonite while the cloud home is still loading, it will start a save for the world before it is done loading.

I think that this causes the unloaded textures/materials to be optimized in a bad way which results in the materials being ruined in the world.

### To Reproduce

(Backup your cloud home before attempting!)

1) Start Resonite
1) Wait until the cloud home begins loading
2) Press X in the top window bar on desktop before the loading is finished (this will not kill the Resonite process, but it will begin syncing to the cloud in the usual empty environment)
3) Some textures may be 'scrambled' the next time you visit the cloud home

### Expected behavior

Triggering a save for a world which has not finished loading, either on purpose or on accident, should not corrupt the assets of the world.

### Screenshots

In these screenshots, these objects did not have the strange wood-like texture before the corruption was triggered.

![2023-10-18 14 42 29](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/14206961/a044a1f4-2629-46ca-a690-613a230fa409)
![2023-10-18 14 42 45](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/14206961/14f6eec5-1a31-4fec-8f33-8914054b5296)
![2023-10-18 14 42 56](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/14206961/7017b75a-f572-47d4-8a3e-468af91b1fe8)

### Resonite Version Number

2023.10.18.449

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Valve Index

### Log Files

[DESKTOP-H976HO2 - 2023.10.18.449 - 2023-10-18 15_37_57.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13030081/DESKTOP-H976HO2.-.2023.10.18.449.-.2023-10-18.15_37_57.log)

### Additional Context

I posted basically the same issue in NeosPublic GitHub https://github.com/Neos-Metaverse/NeosPublic/issues/3458

Also this https://github.com/Neos-Metaverse/NeosPublic/issues/659

### Reporters

Nytra (Discord: nytra)

**Comment by Enverex** [comment]
This is an old Neos issue that has destroyed at least 3 of my worlds over the years.

See https://github.com/Neos-Metaverse/NeosPublic/issues/659

The issue happens if the world is saved before it's completely finished loading. There doesn't appear to be any way to undo the destruction.

**Comment by epicEaston197** [comment]
Yeah I can attest to this my Sun Control Tool and my bow both completely lost their textures

i've since fixed the Sun Control Tool
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76523692/a03a6e7a-76f6-4dda-bb6f-51527855faf8)

**Comment by Nytra** [comment]
This is still happening in Beta 2024.11.12.1329

I just corrupted my cloud home accidentally again

**Comment by Nytra** [comment]
I think here is another instance of this bug happening https://discord.com/channels/1040316820650991766/1367730691122397244/1367730691122397244
